src/main/java/bobby/ui/Bobby.java

```diff
@@ -0,0 +1,366 @@
+ * It can mark tasks as done, unmark, delete and add tasks.
+ */
+public class Bobby {
+
+    /**
+     * Greeting message displayed when the bot starts
+     */
+    private static final String GREETING = "Hello! I'm Bobby\n"
+            + "What can I do for you?\n";
+
+    /**
+     * Exit msg when the bot terminates
+     */
+    private static final String EXIT = "Bye. Hope to see you again soon!";
+
+    /**
+     * Scanner to read user input
+     */
+    private static Scanner scan = new Scanner(System.in);
+
+    /**
+     * Storage instance for saving/loading tasks from a file
+     */
+    private static Storage storage;
+
+    /**
+     * TaskList instance to manage the list of task
+     */
+    private static TaskList taskList;
+
+    /**
+     * Parser instance to handle command parsing
+     */
+    private static Parser parser;
+
+    /**
+     * File path to store task data
+     */
+    private static final String FILE_PATH = "/Users/zhiyi/Desktop/CS2103T/Chatbot/src/main/task.txt";
+
+    /**
+     * Enum representing the basic types of tasks that Bobby can handle.
+     * T - ToDo
+     * D - Deadline
+     * E - Event
+     */
+    private enum SimpleType {
+        T, D, E
+    }
+
+    /**
+     * Enum representation of the action that Bobby can take.
+     */
+    public enum ActionType {
+        bye, list, delete, mark, unmark, todo, deadline, event, retry, find
+    }
+
+    /**
+     * Constructs a Task object from a string description.
+     *
+     * @param desc The string description of the task.
+     * @return A Task object corresponding to the description,
+     *      or null if the input is invalid
+     */
+
+    public static Task constructTask(String desc) {
+        String details[] = desc.split("/");
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy HHmm");
+        switch(SimpleType.valueOf(details[0])) {
+        case T:
+            Task task = new ToDos(details[2]);
+            if (details[1].equals("1")) {
+                task.isDone = true;
+            } else {
+                task.isDone = false;
+            }
+            return task;
+        case D:
+            task = new Deadlines(details[2],
+                    LocalDateTime.parse(details[3], formatter));
+            if (details[1].equals("1")) {
+                task.isDone = true;
+            } else {
+                task.isDone = false;
+            }
+            return task;
+        case E:
+            task = new Events(details[2],
+                    LocalDateTime.parse(details[3], formatter),
+                    LocalDateTime.parse(details[4], formatter));
+            if (details[1].equals("1")) {
+                task.isDone = true;
+            } else {
+                task.isDone = false;
+            }
+            return task;
+        default:
+            System.out.println("GG!! your code is buggy");
```

Comment:
Haha! While I think this is funny, perhaps it would be more useful to have a more descriptive error message i.e. The task type is invalid?